src/cargo/util/toml/mod.rs

```diff
@@ -1595,7 +1595,42 @@ pub struct InheritableFields {
     ws_root: PathBuf,
 }
 
+/// Defines simple getter methods for inheritable fields.
+macro_rules! inheritable_field_getter {
+    ( $(($path:literal, $field:ident -> $ret:ty),)* ) => (
+        $(
+            #[doc = concat!("Gets the field `workspace.", $path, "`.")]
+            pub fn $field(&self) -> CargoResult<$ret> {
+                if let Some(val) = &self.$field {
+                    return Ok(val.clone())
+                }
+                bail!("`workspace.{}` was not defined", $path);
```

Comment:
Nit: I am not sure `path` is the correct name here. It makes me think of a file path more than something within `Cargo.toml`
```suggestion
    ( $(($key:literal, $field:ident -> $ret:ty),)* ) => (
        $(
            #[doc = concat!("Gets the field `workspace.", $key, "`.")]
            pub fn $field(&self) -> CargoResult<$ret> {
                if let Some(val) = &self.$field {
                    return Ok(val.clone())
                }
                bail!("`workspace.{}` was not defined", $key);
```